logstash-core-event-java/src/main/java/com/logstash/Event.java

```diff
@@ -196,13 +196,13 @@ public Event clone()
     }
 
     public String toString() {
-        // TODO: until we have sprintf
-        String host = (String)this.data.getOrDefault("host", "%{host}");
-        String message = (String)this.data.getOrDefault("message", "%{message}");
+        // TODO: (colin) clean this IOException handling, not sure why we bubble IOException here
         try {
-            return getTimestamp().toIso8601() + " " + host + " " + message;
+            return (getTimestamp().toIso8601() + " " + this.sprintf("%{host} %{message}"));
```

Comment:
Style - you use `getTimestamp()` here and then later `this.sprintf(...)` - do you have a preference?